### PR TITLE
Sort file splits in FetchOperator

### PR DESCRIPTION
**Pull request by mbautin**

### CODE REVIEW

**Comment by mbautin** [comment]
Build succeeded: http://jenkins.clearstorydatainc.com/job/hive_pull_requests/80/
@jhartlaub, @markhamstra: please take a look

**Comment by jhartlaub** [comment]
lgtm

**Comment by mbautin** [comment]
@jhartlaub: merge?

File: ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java [hunk 1]
```diff
@@ -254,6 +256,33 @@ private void getNextPath() throws Exception {
...
+  };
+
+  /**
+   * Input splits may not always be sorted by file name, resulting in an invalid order of output
+   * rows. This is fixed by applying stable sorting on file name.
+   */
+  @SuppressWarnings("unchecked")
+  private static void sortInputSplits(InputSplit inputSplits[]) {
+    for (InputSplit inputSplit : inputSplits) {
+      if (!(inputSplit instanceof FileSplit)) {
```
Comment (markhamstra) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java:276]: Log the unexpected?

Comment (mbautin) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java:276]: Actually, we don't expect the split class to always be `FileSplit`. We are just doing this workaround specifically for `FileSplit`s.

Comment (mbautin) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java:276]: E.g. if this is a table backed by HBase, the input format would be different, and we would not have to sort input splits.

Comment (markhamstra) [inline ql/src/java/org/apache/hadoop/hive/ql/exec/FetchOperator.java:276]: Okay, then lgtm

**Pull request merged**